specs/Print.md

```diff
@@ -0,0 +1,425 @@
+                        (errorCode == S_OK) ? L"Print to PDF Stream succeeded"
+                                   : L"Print to PDF Stream failed",
+                        L"Print to PDF Stream");
+                    return S_OK;
+                })
+                .Get()));
+     return true;
+}
+```
+
+```c#
+async void PrintToPdfStream()
+{
+    CoreWebView2PrintSettings printSettings = null;
+    printSettings = WebViewEnvironment.CreatePrintSettings2();
+    printSettings.Orientation = CoreWebView2PrintOrientation.Portrait;
+    printSettings.ShouldPrintBackgrounds = true;
+
+    MemoryStream pdfStream = new MemoryStream();
+    System.IO.Stream stream = await webView.CoreWebView2.PrintToPdfStreamAsync(printSettings);
+    MessageBox.Show(this, "Print to PDF Stream succeeded", "Print To PDF Stream");
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+/// Specifies the print dialog kind.
+[v1_enum] 
+typedef enum COREWEBVIEW2_PRINT_DIALOG_KIND {
+  /// Opens the browser print preview dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW,
+
+  /// Opens the system print dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_SYSTEM_PRINT,
+
+} COREWEBVIEW2_PRINT_DIALOG_KIND;
+
+/// Specifies the printing side.
+[v1_enum] 
+typedef enum COREWEBVIEW2_PRINT_SIDE {
+  /// Single-sided printing.
+  COREWEBVIEW2_PRINT_SIDE_SINGLE,
+
+  /// Double-sided, horizontal printing.
+  COREWEBVIEW2_PRINT_SIDE_BOTH_HORIZONTAL,
+
+  /// Double-sided, vertical printing.
+  COREWEBVIEW2_PRINT_SIDE_BOTH_VERTICAL,
+
+} COREWEBVIEW2_PRINT_SIDE;
+
+[uuid(92269C88-1CC1-47E0-9A87-71458409BA15), object, pointer_default(unique)]
+interface ICoreWebView2_15 : IUnknown {
+  /// Print the current page asynchronously to the specified printer with the provided settings.
+  /// See `ICoreWebView2PrintSettings2` for description of settings. The handler will return `E_ABORT`
+  /// if `printerName` is empty or null or invalid or invalid settings for a given printer.
+  /// 
+  /// The async `PrintWithSettings` operation completes when it finishes 
+  /// printing to the printer. At this time the `ICoreWebView2SPrintWithSettingsCompletedHandler` 
+  /// is invoked. Only one `Printing` operation can be in progress at a time. If
+  /// `PrintWithSettings` is called while a `PrintWithSettings` or `PrintToPdf` or `PrintToPDFStream` 
+  /// job is in progress, the completed handler is immediately invoked with an error.
+  ///
+  /// \snippet AppWindow.cpp PrintWithSettings
+  HRESULT PrintWithSettings(
+    [in] ICoreWebView2PrintSettings2* printSettings,
+    [in] ICoreWebView2PrintWithSettingsCompletedHandler* handler);
+
+  /// Opens the print dialog to print the current page. See `COREWEBVIEW2_PRINT_DIALOG_KIND` 
+  /// for descriptions of print dialog kinds.
+  /// The default value is `COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW`.
+  /// 
+  /// \snippet AppWindow::OpenPrintDialog
+  HRESULT Print([in] COREWEBVIEW2_PRINT_DIALOG_KIND printDialogKind);
```

Comment:
Can this too be an async operation that completes when the printing has completed (like PrintWithSettings?)

Reply:
When print dialog is system print, we don't know whether printing is completed or not. In this case, we don't know when to invoke the completed handler.